src/main/java/uk/gov/companieshouse/accounts/association/controller/AssociationsListForCompanyController.java

```diff
@@ -0,0 +1,60 @@
+package uk.gov.companieshouse.accounts.association.controller;
+
+import java.util.Objects;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.companieshouse.accounts.association.AccountsAssociationServiceApplication;
+import uk.gov.companieshouse.accounts.association.exceptions.BadRequestRuntimeException;
+import uk.gov.companieshouse.accounts.association.service.AssociationsService;
+import uk.gov.companieshouse.api.accounts.associations.api.AssociationsListForCompanyInterface;
+import uk.gov.companieshouse.api.accounts.associations.model.AssociationsList;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+@RestController
+public class AssociationsListForCompanyController implements AssociationsListForCompanyInterface {
+
+    private final AssociationsService associationsService;
+
+    private static final Logger LOG = LoggerFactory.getLogger( AccountsAssociationServiceApplication.applicationNameSpace );
+
+    public AssociationsListForCompanyController(AssociationsService associationsService) {
+        this.associationsService = associationsService;
+    }
+
+    @Override
+    public ResponseEntity<AssociationsList> getAssociationsForCompany( final String companyNumber, final String xRequestId, final Boolean includeRemoved, final Integer pageIndex, final Integer itemsPerPage ) {
+
+        LOG.debug( String.format( "%s: Attempting to fetch users that are associated with company %s. includeRemoved=%b, itemsPerPage=%d, and pageIndex=%d.", xRequestId, companyNumber, includeRemoved, itemsPerPage, pageIndex ) );
+
+        if ( Objects.isNull( xRequestId ) ){
```

Comment:
this is already checked by non null on interface, dont need this to be added.
Test it by removing it and manually running the app locally.

Reply:
Good spot! I have now removed the redundant check.